### DISCUSSION TITLE
Why is the field name not showing correctly?

### DISCUSSION THREAD

**Discussion by shopapps** [noxoua/filament-activity-log#29] [Q&A] [answered]
Hi, this is a great package, I am just setting up and noticed the following:

for a given Logger with field definitions such as:
```php
/**
     * @return Fields\Fields
     */
    public static function fields(Fields\Fields $fields): Fields\Fields
    {
        return $fields->schema([
            'name',
            'tenantReferenceName',

            'created_at' => fn (Field $field) => $field->badge('info')->date('Y-m-d H:i:s'),
            'updated_at' => fn (Field $field) => $field->badge('info')->date('Y-m-d H:i:s'),
        ]);
    }
```

The output prefixes `filament-activity-log::activities.attributes` to the field name in the output table i.e. `filament-activity-log::activities.attributes.tenantReferenceName` and a similar prefix with `filament-activity-log::activities.subjects` in other areas

I'm guessing it has something to do with translations but is there a way to disable this by default as i dont want to translate every single column of every single table in every database to use this  ☺️

<img width="949" alt="Screenshot 2023-11-03 at 16 01 14" src="https://github.com/noxoua/filament-activity-log/assets/8806928/6feae6fb-8918-40fb-b50b-13af09009a7e">

**Comment by noxoua** [comment, marked as answer]
Sorry for the late reply, I had disabled notifications so I didn't notice your message.

I upgraded the package and changed the structure.
Please update the package to the latest version - `composer update`.

Now to display the label correctly you can use this syntax `->label('Tenant Reference Name')`.

Here's an example of your code with the new structure:

```php
public static function resource(ResourceLogger $logger): ResourceLogger
{
    return $logger->fields([
        'name',
        'tenantReferenceName',
        Field::make('created_at')->badge('info')->date('Y-m-d H:i:s'),
        Field::make('updated_at')->badge('info')->date('Y-m-d H:i:s'),
    ]);
}
```

For more information you can check the [📚 Full documentation](https://activity-log.noxo.app/)

**Comment by shopapps** [comment]
wow, some big changes there - nice!

just spent about an hour refactoring my code to get it to work, so all looking good now thanks.  



For the benefit of others if you run composer update it will fail at `postAutoLoadDump` with the error below until you change all the trait references (also listed below).

```
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Symfony\Component\ErrorHandler\Error\FatalError 

  Trait "Noxo\FilamentActivityLog\Concerns\Resource\LogCreateRecord" not found
```

make changes in your resource filament create and update pages

```
use Noxo\FilamentActivityLog\Concerns\Resource\LogCreateRecord;
# change to
use Noxo\FilamentActivityLog\Extensions\LogCreateRecord;

use Noxo\FilamentActivityLog\Concerns\Resource\LogEditRecord;
# change to:
use Noxo\FilamentActivityLog\Extensions\LogEditRecord;
```

make all the changes and then re-run `composer update` and you should be good to go.


## secondary question
A quick related question...  If I add/remove roles/permissions (using spaties permissions package) how can i get it to just display the roles/permissions **names** that were added/removed (or show a list of before and after names would also work for me).  Before it was just showing an array of names that were changed.

currently, if i try:

```php
->fields([
                Field::make('name')->label(__('name')),
                Field::make('email')->label(__('email')),

                Field::make('roles')
                    ->relation()
                    ->label(__('roles')),
                Field::make('permissions')
                    ->relation()
                    ->label(__('permissions')),
```
the output (which is way overkill for me) is something like:
<img width="1709" alt="Screenshot 2023-11-06 at 08 46 10" src="https://github.com/noxoua/filament-activity-log/assets/8806928/35f73dd6-85ba-4c43-8bae-e4d24ee135b2">

where as if i try:

```php
Field::make('roles.name')
                    ->label(__('Roles')),

                Field::make('permissions.name')
                    ->label(__('Permissions')),
```
i get 

<img width="1718" alt="Screenshot 2023-11-06 at 08 45 44" src="https://github.com/noxoua/filament-activity-log/assets/8806928/06b1cf05-108e-419e-84e9-554757408e58">

if i try and edit the above to 
```php
Field::make('roles.name')
                    ->relation()
                    ->label(__('Roles')),

                Field::make('permissions.name')
                    ->relation()
                    ->label(__('Permissions')),
```

i get an error on the User save action of:

```
Call to undefined relationship [name] on model [Spatie\Permission\Models\Role]. 
```

Regards, 

Paul.

**Comment by shopapps** [comment]
to answer my own question i experimented and discovered:

```
Field::make('roles')
                    ->relation('name')
                    ->label(__('Roles')),

                Field::make('permissions')
                    ->relation('name')
                    ->label(__('Permissions')),
```

does what i need, but feels wrong to have the column name in the relation call and the relation name in the make call  - lol

<img width="1664" alt="Screenshot 2023-11-06 at 09 09 48" src="https://github.com/noxoua/filament-activity-log/assets/8806928/9513f3dd-4878-4788-982c-30cbf36c3309">

**Reply by noxoua** [reply]
Glad everything works for you 👍 

You're right, it could be confusing. I plan to improve this part a little later.

But now you can improve readability with `preloadRelations `, so you don't have to specify `relation` on every field.


```php
$logger
	->preloadRelations('roles', 'permissions')
	->fields([
		Field::make('roles.name')->label(__('Roles')),
		Field::make('permissions.name')->label(__('Permissions')),
	])
```

Hope this will be useful 🙌

**Reply by shopapps** [reply]
yeah i had `preloadRelations ` set when i got that error, let me try again just to be sure..  1 sec.

**Comment by shopapps** [comment]
definitely not quite right...

```php
public static function resource(ResourceLogger $logger): ResourceLogger
    {
        return $logger
            ->preloadRelations(
            // Preload your relationships,
            // this is necessary to log the data correctly.
                'roles', 'permissions',
            )
            // Resource Fields
            ->fields([
                Field::make('name'),
                Field::make('email'),

                Field::make('roles.name')
//                    ->relation('name')
//                    ->label(__('Roles'))
                ,

                Field::make('permissions.name')
//                    ->relation('name')
//                    ->label(__('Permissions'))
                ,

```

produces a display output of:

<img width="1714" alt="Screenshot 2023-11-06 at 09 39 16" src="https://github.com/noxoua/filament-activity-log/assets/8806928/7a0ef77a-42fd-4f8f-8816-fa873763a76a">

but weirdly no ` undefined relationship [name] ` error this time when i hit save on the **EditUser** page 🤔

will experiment a little more.

**Reply by noxoua** [reply]
Really weird 🧐  I'm currently working on some minor edits, will launch a new release in about an hour 👨‍💻

**Reply by shopapps** [reply]
I should probably open a different question for this (let me know if that's what you prefer), but is there a prescribed way to add a second date picker to the filters at the top so I can search between a from and to date range? 😇

**Reply by noxoua** [reply]
Good point, I also thought about replacing the date filter with a date range filter. I think I'll be able to implement this in the next few days.

**Reply by noxoua** [reply]
The new version is already available. Now everything should work properly 😊

```php
$logger->fields([
   Field::make('roles.name')
         ->hasMany('roles')
         ->label(__('Roles'))
         ->badge(),

   Field::make('permissions.name')
         ->hasMany('permissions')
         ->label(__('Permissions'))
         ->badge(),
])
```

<img width="1087" alt="relation-screenshot" src="https://github.com/noxoua/filament-activity-log/assets/142592979/5a24c91e-445c-4d27-9984-c20617c7f0f6">

**Reply by shopapps** [reply]
awesome.. good job, I'm just in a call so will install and test shortly 👍

**Reply by shopapps** [reply]
can confirm the above works well - thank you for the outstanding work 😎

**Comment by shopapps** [comment]
> Really weird 🧐 I'm currently working on some minor edits, will launch a new release in about an hour 👨‍💻

whist you are working on this, i can't seem to change the title of the page and navigation link title:

```
class Audits extends ListActivities
{
    protected static ?string $navigationIcon = 'tabler-help';


    protected static ?string $navigationGroup = 'Administration';
    protected static ?string $navigationLabel = 'Audit History';
    protected static ?string $title = 'Audit History';
}
```

does nothing.  Title remains: `Activity history`

**Reply by noxoua** [reply]
Since translations are used by default, you need to override `getTitle` and `getNavigationLabel`.

```php
    public function getTitle(): string
    {
        return __('filament-activity-log::activities.title');
    }

    public static function getNavigationLabel(): string
    {
        return __('filament-activity-log::activities.title');
    }

```

**Reply by shopapps** [reply]
I saw that thanks..  For the benefit of others, using a more filament-esque approach you can use something like:

```php

<?php

namespace App\Filament\Pages;

use Noxo\FilamentActivityLog\Pages\ListActivities;

class Audits extends ListActivities
{
    protected static ?string $navigationIcon = 'tabler-help';


    protected static ?string $navigationGroup = 'Administration';
    protected static ?string $navigationLabel = 'Audit History';
    protected static ?string $title = 'Audit History';

    public function getTitle(): string
    {
        return __(static::$title ?? (string) str(class_basename(static::class))
            ->kebab()
            ->replace('-', ' ')
            ->title()
        );
    }

    public static function getNavigationLabel(): string
    {
        return __(static::$navigationLabel ??
            static::$title ??
            __('filament-activity-log::activities.title')
        );
    }
    
}
```
